tensorbay/cli/log.py

```diff
@@ -387,16 +505,27 @@ def _combine_details(
         for prefix, message in zip_longest(merge_prefixes, messages, fillvalue=fillvalue):
             yield f"{prefix}  {message}\n"
 
-    @staticmethod
-    def _build_commit_tree(
-        dataset_client: DatasetClientType, revisions: List[Optional[str]]
-    ) -> List[_CommitNode]:
+    def _build_tree(
+        self,
+        dataset_client: DatasetClientType,
+        revisions: List[Optional[str]],
+        show_drafts: bool,
+    ) -> List[_T]:
         commit_to_node: Dict[str, _CommitNode] = {}
-        leaves: Dict[str, _CommitNode] = {}
+        leaves: Dict[str, _T] = {}
         for revision in revisions:
-            commits = dataset_client.list_commits(revision)
-            child_node: Optional[_CommitNode] = None
-            for index, commit in enumerate(commits):
+            child_node: Optional[_T] = None
+            if show_drafts:
+                drafts = dataset_client.list_drafts(branch_name=revision)
+                if drafts:
+                    draft = drafts[0]
+                    draft_node = _DraftNode(draft)
+                    key = draft_node.key
+                    leaves[key] = draft_node
+                    self._key_to_branches[key] = [draft.branch_name]
+                    child_node = draft_node
```

Comment:
The latest commit on the branch may not be the parent commit of the draft.
Use `Draft.parent_commit_id` to find the parent node.